blobfs/src/lib.rs

```diff
@@ -168,6 +169,20 @@ pub struct BlobFs {
 }
 
 impl BlobFs {
+    fn ensure_path_exist(path: &Path) -> io::Result<()> {
+        if path.as_os_str().is_empty() {
+            return Err(einval!("path is empty"));
+        }
+        if !path.exists() {
+            create_dir_all(path).map_err(|e| {
+                error!("create dir error: {:?}", path);
+                e
```

Comment:
can we use use macro einval!(e) which shows more verbose message including file!() and line!()

Reply:
macro einval! is more convenient, but it will lose real error code return from create_dir_all().

Reply:
I'm fine with it, but we need to make `error!("create dir error: {:?}", path);` more verbose so that we know where goes wrong exactly.